jdaviz/configs/imviz/helper.py

```diff
@@ -79,6 +87,89 @@ def load_data(self, data, parser_reference=None, **kwargs):
             self.app.load_data(
                 data, parser_reference=parser_reference, **kwargs)
 
+    def center_on(self, point):
+        """Centers the view on a particular point.
+
+        Parameters
+        ----------
+        point : tuple or `~astropy.coordinates.SkyCoord`
+            If tuple of ``(X, Y)`` is given, it is assumed
+            to be in data coordinates and 0-indexed.
+
+        Raises
+        ------
+        AttributeError
+            Sky coordinates are given but image does not have a valid WCS.
+
+        """
+        viewer = self.app.get_viewer("viewer-1")
+
+        if isinstance(point, SkyCoord):
+            i_top = get_top_layer_index(viewer)
+            image = viewer.layers[i_top].layer
```

Comment:
I think you should probably use image = viewer.reference_data as that is the one which is used to define the pixel space

Reply:
Your suggestion does not seem to work: `AttributeError: 'ImvizImageView' object has no attribute 'reference_data'`

Reply:
`viewer.state.reference_data` exists but it causes my test to fail:

```
        # Blink to the one without WCS
        self.viewer.blink_once()
    
        with pytest.raises(AttributeError, match='does not have a valid WCS'):
>           self.imviz.center_on(sky)
E           Failed: DID NOT RAISE <class 'AttributeError'>

jdaviz/configs/imviz/tests/test_astrowidgets_api.py:83: Failed
```

Reply:
It appears `reference_data` does not keep track of which layer is actually visible to the user after blinking.